lib/_stream_writable.js

```diff
@@ -379,10 +379,18 @@ function clearBuffer(stream, state) {
     holder.entry = entry;
 
     var count = 0;
-    while (entry) {
-      buffer[count] = entry;
-      entry = entry.next;
-      count += 1;
+    if(state.objectMode){
```

Comment:
@nrvru please run `make lint`, and then address any errors. You should have seen errors in `make test`.